File: Intersection of Two Arrays.go
func intersection(nums1 []int, nums2 []int) []int {
    dp := make(map[int]interface{})
    for _, num := range nums1 {
        dp[num] = nil
    }

    arr := make([]int, 0)
    for _, num := range nums2 {
        if _, ok := dp[num]; ok {
            arr = append(arr, num)
            delete(dp, num)
        }
    }
    return arr
}

File: Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> n1(nums1.begin(), nums1.end());
        set<int> n2(nums2.begin(), nums2.end());
        vector<int> ans;
        for (int num: n1) {
            if (n2.contains(num)) ans.push_back(num);
        }
        return ans;
    }
};
